feat(version): add --premajor-version-bump option to force patch bumps for non-breaking changes in premajor packages (#3876)
lerna/lerna: libs/commands/version/README.md

When run with this flag, `lerna version` will increment `premajor`, `preminor`, `prepatch`, or `prerelease` semver
bumps using the specified [prerelease identifier](http://semver.org/#spec-item-9).

### `--premajor-version-bump`

This option allows you to control how lerna handles bumping versions for `premajor` version release packages. 

> By default, this is configured as `"semver"`, and any `premajor` package is bumped as `minor`

Reviewer: The "any premajor package is bumped as minor" is not true. Depending on the changelog-preset, fixes will still be bumped as patches. It's just that non-breaking features trigger minor bumps.

```suggestion
By default, this is configured as `"semver"`. This option only affects non-breaking changes in `premajor` packages. Breaking changes in `premajor` packages will always trigger a `minor` bump.
```

```sh
lerna version --conventional-commits --premajor-version-bump semver

> # ensures premajor version bumps are handled as minor unless breaking change is applied.

Reviewer: ```suggestion
# allows for non-breaking changes to trigger either a minor or patch bump as specified by the changelog-preset
```

Author: Hmm, sorry, it could be that I got something wrong when making the feature request if it is possible to adjust the way this is handled using `changelog-preset`. 

When I looked at the code it seemed to me that the `changelog-preset` and the recommended bump from that comes through `conventionalRecommendedBump` on line 59 in `recommend-version.ts` while `if (semver.major(pkg.version) === 0) {` overrides the recommended bump inside of the block starting on line 77 ?

This was originally why I ended up requesting this feature, since I couldn't find any way to trigger it using `changelog-preset`. 🤔 

Did I get this totally wrong ?

Author: I've rephrased this as: `# ensures that all feature version bumps are handled as minor`. If changelog preset is configured so that a `premajor version` `feat` commit is treated as `patch` the same feature can be achieved but it seems like the expected behaviour in currently all changelog-presets handle this as `minor`.

# 0.1.0 --> 0.2.0

lerna version --conventional-commits --premajor-version-bump node-semver

> # ensures premajor version bumps are handled as patches unless breaking change is applied.

Reviewer: ```suggestion
# ensures that all non-breaking premajor version bumps are handled as patches
```
